### DISCUSSION TITLE
What is a pull_request synchronize event

### DISCUSSION THREAD

**Discussion by simonbyrne** [community/community#24567] [Apps, API and Webhooks]
<p>What precisely causes a synchronize event? The <a href="https://developer.github.com/v3/activity/events/types/#pullrequestevent" rel="nofollow noopener">docs</a> don’t define what it is.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/simonbyrne">@simonbyrne</a>,</p>
<p>Thank you for being here! A pull_request event it’s only triggered when the pull request’s tracking branch is synchronized with the source branch for the pull request, and that happens when the source branch is updated.</p>

**Comment by simonbyrne** [comment]
<p>Hi Andrea,</p>
<p>Thanks for your reply. Do you know if it also gets triggered when the base branch (that the PR is merging into) is updated?</p>
<p>Thanks,</p>
<p>Simon</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hey <a class="mention" href="/simonbyrne">@simonbyrne</a>,</p>
<p>Thanks for the follow up. What this implies is that if you push a new commit to the HEAD ref of a pull request then this “synchronize” event will be triggered. This is because the system is syncing the pull requests with the latest changes. This will NOT trigger if the base ref is updated.</p>

**Reply by carltonbrown** [reply]
>  This will NOT trigger if the base ref is updated.

I wonder if this info has changed in the past few years, since [current Github docs](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request) appear to say something different:

> synchronize: Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or **when the base branch is changed.**

**Reply by mkuijn-cb** [reply]
When the base branch is changed means: when you change the base ref (event "edited" on pull_request)

When there are pushes to the branch the PR's base ref currently points to, the "synchronize" event is *not* triggered.

**Reply by ventsyv** [reply]
> What this implies is that if you push a new commit to the HEAD ref of a pull request then this “synchronize” event will be triggered.

Wouldn't that also trigger on.push event? For example if I do the following:
```
git checkout myBranch
git pull origin master
git push
```

If myBranch has an open pull request against the master branch, should I expect on.push, on.pull_request.synchronize, or both ?

**Reply by grigorye** [reply]
@ventsyv If I get it correctly, you'll get both events. So if you want to avoid *triggering two workflow runs* on push (if you register for both kinds of events), then you would need to employ some filters, e.g. like below:

```
on:  
  push:
    branches: [main] # Only pushes to main branch trigger the workflow.

  pull_request: {} # Pull request events (default: open, synchronized, reopened) in any branch triggers the workflow.
```

Just in case, the two workflow runs would still be different because of different contexts associated with push and pull_request events.

**Reply by siloKalle** [reply]
I have a monorepo where I would like to use paths as filter the specific workflows for the certain folders. But the problems is that using paths parameter does not have any affect at all. It seems that the synchronized is ruining it. 

Why the documentation says that you can do that although I cannot acltually use pull_request finetune based on the "paths" parameter? 

I can do it with push event, but not with pull request trigger event.

**Reply by kasvith** [reply]
im also facing another issue related to this, if i open a pr to a branch and(main to prod), and if this pr fails to build it also affect the status of base branch(main)

**Reply by socketbox** [reply]
@AndreaGriffiths11 What would you think of a suggestion to update the GitHub Actions workflow documentation regarding event types such that there are links from each event type name to the description for that event type in the WebHooks documentation?

**Comment by Nearoo** [comment]
How come this question is asked in 2020 and there is still no documentation about it?

**Comment by carltonbrown** [comment]
> How come this question is asked in 2020 and there is still no documentation about it?

Hello,

If you navigate to the [pull_request event documentation](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=synchronize#pull_request) and click the `Action type` dropdown, you can select the `synchronize` action type and read the documentation, which states:

> A pull request's head branch was updated. For example, the head branch was updated from the base branch or new commits were pushed to the head branch.

Every push to a repository will generate a `push` event.  A `synchronize` event is generated when the PR metadata is updated with the details of the push (these are asynchronous steps in GitHub).

If you want a workflow that responds to changes in the PR merge outcome, then you should subscribe to the `synchronize` event.   There is a risk of executing the workflow twice if it subscribes to `synchronize` and `push`.   But if your workflow monitors pushes that aren't connected to a PR, or aren't interested in PR state, then use only the `push` event.

**Reply by sergiubologa** [reply]
I was there multiple times and didn't see the dropdown. I had to find this comment to find it 😄 
![image](https://github.com/community/community/assets/10432669/bf448e14-3b48-4302-9c76-a763b65f3279)

**Reply by sagalovsky** [reply]
> If you want a workflow that responds to changes in the PR merge outcome, then you should subscribe to the synchronize event. There is a risk of executing the workflow twice if it subscribes to synchronize and push.

Should the first sentence say `synchronize` or `push`?

Based on
> A pull request's head branch was updated. For example, the head branch was updated from the base branch or new commits were pushed to the head branch.

the `synchronize` event is not triggered when the PR merge outcome changes due to new commits being pushed to the base branch.

**Reply by Joehl1103** [reply]
This did it for me. Thank you! Please make this clearer or searchable in the documentation!

**Comment by SerkanSipahi** [comment]
@carltonbrown 

Maybe it shouldn't be hidden behind a dropdown. It was very difficult to find.
